boa3_test/tests/compiler_tests/test_range.py

```diff
@@ -62,30 +60,20 @@ def test_range_given_length(self):
             + Opcode.RET        # return
         )
 
-        path = self.get_contract_path('RangeGivenLen.py')
-        output = self.compile(path)
+        output, _ = self.assertCompile('RangeGivenLen.py')
         self.assertEqual(expected_output, output)
 
-        path, _ = self.get_deploy_file_paths(path)
-        runner = BoaTestRunner(runner_id=self.method_name())
-
-        invokes = []
-        expected_results = []
-
-        invokes.append(runner.call_contract(path, 'range_example', 5))
-        expected_results.append([0, 1, 2, 3, 4])
-        invokes.append(runner.call_contract(path, 'range_example', 10))
-        expected_results.append([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
-        invokes.append(runner.call_contract(path, 'range_example', 0))
-        expected_results.append([])
-
-        runner.execute()
-        self.assertEqual(VMState.HALT, runner.vm_state, msg=runner.error)
+    async def test_range_given_length(self):
+        await self.set_up_contract('RangeGivenLen.py')
 
-        for x in range(len(invokes)):
-            self.assertEqual(expected_results[x], invokes[x].result)
+        result, _ = await self.call('range_example', [5], return_type=list)
+        self.assertEqual([0, 1, 2, 3, 4], result)
+        result, _ = await self.call('range_example', [10], return_type=list)
+        self.assertEqual([0, 1, 2, 3, 4, 5, 6, 7, 8, 9], result)
+        result, _ = await self.call('range_example', [0], return_type=list)
+        self.assertEqual([], result)
```

Comment:
I know this is how the tests were before, but since we are refactoring this test case anyway, could you change the expected results to `list(range(args))`?
```suggestion
        result, _ = await self.call('range_example', [5], return_type=list)
        self.assertEqual(list(range(5)), result)
        result, _ = await self.call('range_example', [10], return_type=list)
        self.assertEqual(list(range(10)), result)
        result, _ = await self.call('range_example', [0], return_type=list)
        self.assertEqual(list(range(0)), result)
```